### PR TITLE
Add loading screen on initial page load

### PR DESCRIPTION
**Pull request by ptbarnum4**
@harveysanders I would like your opinion on adding a loading page

- Substantially reduces initial load time + adds a smoother initial loading experience 
- Only occurs upon initial page load or refresh

### CODE REVIEW

File: pages/_app.tsx [hunk 1]
```diff
@@ -23,22 +25,25 @@ export default function App({ Component, pageProps }: AppProps) {
       await getStaticAsset('logos').then((l: ISupporterFunderLogos) => setLogos(l.funders));
       await getStaticAsset('alert').then(setAlertInfo);
     };
-    fetchStaticAssets();
+    fetchStaticAssets().then(() => setTimeout(() => setLoading(false), 1000));
```
Comment (harveysanders) [inline pages/_app.tsx:28]: Why do you need to wait another second to toggle the loading state?
```suggestion
    fetchStaticAssets().then(() => setTimeout(() => setLoading(false), 1000));
```

**Review by harveysanders — COMMENTED** [review]
I'm cool with the loading page, but why the 1 second`setTimeout`?

**Comment by harveysanders** [comment]
Actually, why do we need a loading page at all? I thought all (or most) of this site is pre-built as static files and ready to go.

**Comment by harveysanders** [comment]
Yeah, sorry. The more I think about it, the more I dislike the idea of this loading page for a static site. 
The loading screen itself looks great. I'm just really opposed to the idea of it for this info site. It would be another thing if this were more of an application where we have to load many heavy assets.  I think our efforts are better spent figuring out why the initial load is slow and fixing that.

**Comment by ptbarnum4** [comment]
we don't. I just thought it looked cool + instant load

image

**Comment by ptbarnum4** [comment]
the initial load isn't slow, and the overall load time is barely different with or without. Instead of 800ms of transition time where the page is being sent over is replaced by the loading screen. 

I wasn't betting on it being added, just thought I would throw out the idea.

File: pages/_app.tsx (hunk 1 above)
Comment (ptbarnum4) [reply pages/_app.tsx:28]: that was so it didn't flash awkwardly. it really only shows for 1000ms (+ time to send over the loading page <100ms)

**Comment by harveysanders** [comment]
Ah got you. Well, it does look cool... I just think it's out of place for a static site.

[This](https://motherfuckingwebsite.com/) comes to mind. If you haven't seen it before, it's of course satire, but there is a lot of truth on that site.

**Pull request closed without merging**